core/server/services/email-analytics/lib/event-processor.js

```diff
@@ -136,10 +137,30 @@ class GhostEventProcessor extends EventProcessor {
             return false;
         }
 
+        const subscribedNewsletterIds = (
+            await this.db.knex('members_newsletters')
+                .where('member_id', '=', memberId)
+                .select('newsletter_id')
+        ).map(mn => mn.newsletter_id);
+
+        await this.db.knex('members_newsletters')
+            .where('member_id', '=', memberId)
+            .del();
+
+        for (const newsletterId of subscribedNewsletterIds) {
+            await this.db.knex('members_subscribe_events').insert({
+                id: ObjectID().toHexString(),
+                member_id: memberId,
+                newsletter_id: newsletterId,
+                subscribed: false,
+                created_at: moment.utc().toDate(),
```

Comment:
Will these all be exactly the same? If not, it would be nice to generate it once, and use the same timestamp for all events.